internal/gardener/gardener.go

```diff
@@ -24,6 +26,7 @@ const (
 
 type gardenerProvisioner struct {
 	operator operator.Operator
+	timeouts types.Timeouts
```

Comment:
Options should be passed when calling Hydroform functions, not part of the provisioner.
See:
https://github.com/kyma-incubator/hydroform/blob/97160354ef3e83947e323214aeb8d93f57d2fd92/examples/gardener/gcp/main.go#L69

Reply:
It is passed to the Hydroform function see here https://github.com/kyma-incubator/hydroform/pull/53/files#diff-defc0a0ee6263735b44fdcc9504d921bR38 only later the timeout are assigned to the provider in the `New` function here https://github.com/kyma-incubator/hydroform/pull/53/files#diff-dfa71d4200e54a500c09fed237cbbecdR49 but they come from options the same way as `Persistent()` and ` WithDataDir(dir string)`. Here is a full snippet:
```go
// New creates a new instance of gcpProvisioner.
func New(operatorType operator.Type, ops ...types.Option) *gcpProvisioner {
	// parse config
	os := &types.Options{}
	for _, o := range ops {
		o(os)
	}

	var op operator.Operator
	switch operatorType {
	case operator.TerraformOperator:
		tfOps := terraform_operator.ToTerraformOptions(os)
		op = terraform_operator.New(tfOps...)
	default:
		op = &operator.Unknown{}
	}

	return &gcpProvisioner{
		provisionOperator: op,
		timeouts:          os.Timeouts,
	}
}
```
And it can be called like that:
```go
	cluster, err := hf.Provision(cluster, provider, types.WithTimeouts(types.Timeouts{Create: 30*time.Minute}))
```

Reply:
Ah I see, then the approach on how the option is defined and given to hydroform is right.

The real issue is assigning it to the provisioners, which is already on our other thread.